File: frontend/src/types/PortfolioModel.ts
import AssetModel from './AssetModel';
import FractionModel from './FractionModel';
import KeyValueModel from './KeyValueModel';
import TransactionModel from './TransactionModel';

interface PortfolioModel {
  id: number;
  name: string;
  totalInvested: number;
  holdingsValue: number;
  totalReturn: number;
  assets: AssetModel[];
  allocation: FractionModel[];
  transactions: TransactionModel[];
  totalInvestedTimeline: KeyValueModel<Date, number>[];
  holdingsValueTimeline: KeyValueModel<Date, number>[];
}

export default PortfolioModel;

File: backend/src/DCAFlow.Contracts/Models/PortfolioModel.cs
namespace DCAFlow.Contracts.Models;

public class PortfolioModel
{
    public int Id { get; set; }

    public string Name { get; set; }

    public double TotalInvested { get; set; }

    public double HoldingsValue { get; set; }

    public double TotalReturn { get; set; }

    public List<AssetModel> Assets { get; set; }

    public List<FractionModel> Allocation { get; set; }

    public List<KeyValueModel<DateTime, double>> Investments { get; set; }
}
